packages/driver/cypress/e2e/commands/sessions/sessions.cy.ts

```diff
@@ -99,31 +105,52 @@ describe('cy.session', { retries: 0 }, () => {
       it('clears the browser cookie after each run', () => {
         cy.window()
         .then((win) => {
-          // @ts-expect-error TODO: cookie doesn't exist on win, usually cookie is set on document
-          win.cookie = 'key=value; SameSite=Strict; Secure; Path=/fixtures'
+          // Check that the only cookie present is the Cypress initial cookie
+          expect(win.document.cookie).to.satisfy((cookie) => {
+            return !cookie || cookie === '__cypress.initial=true'
+          })
+
+          win.document.cookie = 'key=value; SameSite=Strict; Secure; Path=/fixtures'
         })
         .then(async () => {
           cy.spy(Cypress, 'action').log(false)
 
-          await Cypress.action('runner:test:before:after:run:async', {}, Cypress.state('runnable'), { nextTestHasTestIsolationOn: true })
+          await Cypress.action('runner:test:before:after:run:async', {}, Cypress.state('runnable'), {
+            nextTestHasTestIsolationOn: true,
+          })
         })
 
-        cy.window().its('cookie').should('be.undefined')
+        cy.window().then((win) => {
+          // Check that the only cookie present is the Cypress initial cookie
+          expect(win.document.cookie).to.satisfy((cookie) => {
+            return !cookie || cookie === '__cypress.initial=true'
+          })
+        })
       })
 
       it('does not clear the browser cookie after each run if the next test has test isolation off', () => {
         cy.window()
         .then((win) => {
-          // @ts-expect-error TODO: cookie doesn't exist on win, usually cookie is set on document
-          win.cookie = 'key=value; SameSite=Strict; Secure; Path=/fixtures'
+          // Check that the only cookie present is the Cypress initial cookie
+          expect(win.document.cookie).to.satisfy((cookie) => {
+            return !cookie || cookie === '__cypress.initial=true'
+          })
+
+          win.document.cookie = 'key=value'
```

Comment:
This test doesn't pass if you pass the `SameSite=Strict; Secure; Path=/fixtures` to this cookie. I'm not sure if this is some kind of bug, but that's the actual behavior.

Reply:
something like this though I think fundamentally changes the test. I can look at it tomorrow?

Reply:
Well, it wasn’t really testing anything to begin with. :/

Reply:
yeah `win.cookie` really isn't anything 😂 . Even if it was `document.cookie` that wouldn't even be set in the first place because its not even at that path, so it will not set it and no-op. I think this change is fine.